### PR TITLE
bugfix for inconsistent rotation calculation order.

### PR DESCRIPTION
**Pull request by Jackie9527**
## What does this PR do?
1. Open Editor.
2. Click Tools -> Track View.
3. Create a sequence.
4. Add some nodes.
5. Record a key frame, modify the rotation of the entity.
image

## How was this PR tested?
Follow the same steps, as follows:
image

### CODE REVIEW

**Review by gadams3 — APPROVED** [review]
Even though these are “legacy”, are there any tests for the functions? At some point, everything should be migrated to the AZ math types.

File: Code/Legacy/CryCommon/Cry_Quat.h [hunk 1]
```diff
@@ -395,6 +395,30 @@ struct Quat_tpl
         return q;
     }
 
+    // creatr rotation quatation base on ZYX axis
```
Comment (AMZN-daimini) [inline Code/Legacy/CryCommon/Cry_Quat.h:398]: nit: "create rotation quaternion based on ZYX axis", I guess? :)

**Review by AMZN-daimini — APPROVED** [review]
Same as what Guthrie said, it would make sense to look into moving this to use AZ math types over time. No need to block this tho.

File: Code/Legacy/CryCommon/Cry_Quat.h [hunk 2]
```diff
@@ -395,6 +395,30 @@ struct Quat_tpl
...
+    ILINE void SetRotationZYX(const Ang3_tpl<F>& a)
```
Comment (hultonha) [inline Code/Legacy/CryCommon/Cry_Quat.h:399]: update: If we could look at AZ::Quaternion and see if this functionality is already supported could we use that instead (we are long-term planning to deprecate and remove CryMath so adding more functionality should ideally be avoided if possible

Comment (hultonha) [reply Code/Legacy/CryCommon/Cry_Quat.h:399]: We could do this as a follow-up PR if needed, but might be nice to do a pass over Track View (with tests) to update the use of the math library (this is of course a large change though, so might need some thought/planning).

**Review by cgalvan — APPROVED** [review]

**Pull request merged**